File: app/src/main/java/com/czbix/v2ex/eventbus/BaseEvent.java
package com.czbix.v2ex.eventbus;

public abstract class BaseEvent {
    public static class GetNodesFinishEvent extends BaseEvent {}
    public static class NewUnreadEvent extends BaseEvent {
        public final int mCount;

        public NewUnreadEvent(int count) {
            mCount = count;
        }

        public boolean hasNew() {
            return mCount > 0;
        }
    }

    public static class DailyAwardEvent extends BaseEvent {
        public final boolean mHasAward;

        public DailyAwardEvent(boolean hasAward) {
            mHasAward = hasAward;
        }
    }

    public static class ContextInitFinishEvent extends BaseEvent {}
}

File: app/src/main/kotlin/com/czbix/v2ex/event/BaseEvent.kt
package com.czbix.v2ex.event

abstract class BaseEvent {
    class NewUnreadEvent(val mCount: Int) : BaseEvent() {
        fun hasNew(): Boolean {
            return mCount > 0
        }
    }

    class DailyAwardEvent(val mHasAward: Boolean) : BaseEvent()

    class ContextInitFinishEvent : BaseEvent()
}
